### PR TITLE
Add client Brotli support

### PR DESCRIPTION
**Pull request by derlih**
## What do these changes do?

If brotlipy is installed, client will support this compression type

## Are there changes in behavior for the user?

No

## Related issue number

#5219 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Review by asvetlov — CHANGES_REQUESTED** [review]
The PR looks good.

Please click on https://cla-assistant.io/aio-libs/aiohttp?pullRequest=5227 to sign CLA.
I've added it recently and checked -- the sign is pretty easy :)

**Comment by derlih** [comment]
@asvetlov 
Is aiohttp somehow connected to SAP?

**Comment by asvetlov** [comment]
Oops.
No, sure! Thanks for the note.
I've updated the CLA text to closely follow the Python CLA (https://www.python.org/psf/contrib/contrib-form/) but slightly adapted to the aio-libs project.
The CLA text is available here: https://gist.github.com/asvetlov/dbce6679052f5c89d3d13f501107c1d4

**Comment by derlih** [comment]
Signed.
Should the brotli runtime support be reflected in somewhere in docs?

**Comment by asvetlov** [comment]
> Should the brotli runtime support be reflected in somewhere in docs?

Sounds good.  
Do you have an idea of what place the sentence should be pushed in?

**Comment by derlih** [comment]
What about the dependencies section?
https://github.com/aio-libs/aiohttp/blob/021668ad4e2ac4eba1693bd5c5b952911426d908/docs/index.rst#L146-L170

**Comment by asvetlov** [comment]
ok

File: docs/glossary.rst [hunk 1]
```diff
@@ -27,6 +27,14 @@
 
       https://pypi.python.org/pypi/asyncio/
 
+   brotlipy
```
Comment (asvetlov) [inline docs/glossary.rst]: FYI we switched from `brotlipy` to `brotli` a year ago https://github.com/aio-libs/aiohttp/pull/3803

Do you have a reason for the reverse direction?

**Review by asvetlov — CHANGES_REQUESTED** [review]

File: docs/glossary.rst (hunk 1 above)
Comment (derlih) [reply docs/glossary.rst]: brotlipy was mentioned in #5219 
But actually it checked against Brotli package.
https://github.com/derlih/aiohttp/blob/a58b7ed4a4d85a9e5d530e3c99c685455bae65e4/aiohttp/http_parser.py#L29-L34

**Review by asvetlov — APPROVED** [review]

**Review by asvetlov — CHANGES_REQUESTED** [review]
Ooops, the spell checker doesn't approve the docs changes: https://github.com/aio-libs/aiohttp/pull/5227/checks?check_run_id=1400046288

The most practical solution is extending https://github.com/aio-libs/aiohttp/blob/master/docs/spelling_wordlist.txt with words like `brotli` to teach the checker new facilities

**Comment by derlih** [comment]
I fixed the spellcheck.

**Review by asvetlov — APPROVED** [review]

**Comment by asvetlov** [comment]
The feature is backward compatible.
But it is still a new feature, there is no need to backport it to 3.7

**Comment by derlih** [comment]
The aiohttp user from #5219  wanted to use it. May be for 3.7.3?
I think this release will happen quite soon because the win32 wheels are currently missing.

**Comment by asvetlov** [comment]
Sorry, I had no capacity to push it into 3.7.3.
Technically this is a new backward-compatible feature.
It means that it can be released as a part of 3.8.0 pretty soon.
I want to review recent reproducible bug reports (and fix them if possible) first to include fixes into 3.8

**Comment by asvetlov** [comment]
Thanks!

**Comment by ddelange** [comment]
hi 👋 

it seems this patch didn't make it into 3.8.3, the Accept-Encoding header is still [hard-coded](https://github.com/aio-libs/aiohttp/blob/v3.8.3/aiohttp/client_reqrep.py#L232).

could this patch be cherry-picked into 3.8.4?

**Comment by webknjaz** [comment]
You can try following bot instructions to submit a PR against 3.9.

**Comment by ddelange** [comment]
opened backport PRs against 3.9 (#7100) and 3.8 (#7101)

**Comment by ddelange** [comment]
hi 👋 is there a rough ETA when 3.9 and/or 4.0 will be shipped?

**Pull request merged**